lib/msf/core/exploit/http/client.rb

```diff
@@ -44,7 +44,9 @@ def initialize(info = {})
         OptString.new('DOMAIN', [ true, 'The domain to use for windows authentification', 'WORKSTATION']),
         OptFloat.new('HttpClientTimeout', [false, 'HTTP connection and receive timeout']),
         OptBool.new('HttpPartialResponses', [false, 'Return partial HTTP responses despite timeouts', false]),
-        OptBool.new('HttpTrace', [false, 'Show the raw HTTP requests and responses', false])
+        OptEnum.new('HttpTrace', [false, 'Show the raw HTTP requests and responses', 'false', ['true', 'false', 'color']]),
```

Comment:
Maybe a dumb question, what are your thoughts on making logs colored by default when HttpTrace is set to true? Or is that problematic?

Reply:
Logs?

Reply:
Oh, if you mean the `HttpTrace` output, not all terminals support color, so having an option without color is desirable. Also, some people don't want color.

This also preserves existing behavior to a degree. That said, the `OptEnum` with Boolean-like values is a total hack of the command dispatcher.

If the `color` option is set on a terminal that doesn't support color, the console should handle that just fine, even with `spool`. We shouldn't need to handle that here. Even `Rex::Text::Color` is _somewhat_ terminal-conscious.

Reply:
Oh yeah, the other use case I was thinking about was copying and pasting `HttpTrace` output to GitHub, etc. You'd have to screenshot to get the colors, and you can't copy and paste the request or response from that (OCR notwithstanding).

Reply:
If we want to change the long-standing behavior of `HttpTrace`, maybe we can default to color and then have an colorless option? I opted not to change existing behavior too much, but I don't think this option sees much use. I greatly prefer the color!

Reply:
If we want to change the long-standing behavior of `HttpTrace`, maybe we can default to color and then have a colorless option? I opted not to change existing behavior too much, but I don't think this option sees much use. I greatly prefer the color!

Reply:
If we can come up with good names for the options, I'm happy to make the change to the default.

Reply:
After a little discussion, we decided to do color and show the banners, so we get the best of both worlds. The color will grab the eye, anyway. The underlying code should handle non-colorized terminals just fine. 💪

Reply:
Added benefit of showing both is that newbies know which is the request and which is the response.

Reply:
This actually looks really rad now with both. Thanks, @adfoster-r7!

Reply:
It also offers a clearer delineation between request and response when no response is received.